bin/node/src/commands/block_producer.rs

```diff
@@ -73,29 +73,25 @@ impl BlockProducerCommand {
             batch_interval,
         } = self;
 
-        let store_url = store_url
+        let store_address = store_url
             .to_socket()
             .context("Failed to extract socket address from store URL")?;
 
-        let listener =
-            url.to_socket().context("Failed to extract socket address from store URL")?;
-        let listener = tokio::net::TcpListener::bind(listener)
-            .await
-            .context("Failed to bind to store's gRPC URL")?;
+        let block_producer_address = url
+            .to_socket()
+            .context("Failed to extract socket address from block producer URL")?;
 
-        BlockProducer::init(
-            listener,
-            store_url,
-            batch_prover_url,
-            block_prover_url,
+        BlockProducer {
+            block_producer_address,
+            store_address,
+            batch_prover: batch_prover_url,
+            block_prover: block_prover_url,
```

Comment:
nit: I think these fields could be named `BlockProducer::batch_prover_url` and `BlockProducer::block_prover_url` to make them a bit more consistent